tests/tests.rs

```diff
@@ -1470,3 +1470,186 @@ fn test_base_directory() {
         ),
     );
 }
+
+/// --prune
+#[test]
+fn test_prune() {
+    let dirs = &["foo", "bar/zap/foo"];
+    let files = &[
+        "foo/a.foo",
+        "foo/b.bar",
+        "bar/zap/c.foo",
+        "bar/zap/foo/d.foo",
+    ];
+    let te = TestEnv::new(dirs, files);
+
+    // without pruning
+    te.assert_output(
+        &["foo"],
+        "bar/zap/foo/d.foo
+         bar/zap/c.foo
+         bar/zap/foo
+         foo/a.foo
+         foo",
+    );
+    // with pruning
+    te.assert_output(
+        &["--prune", "foo"],
+        "bar/zap/c.foo
+         bar/zap/foo
+         foo",
+    );
+
+    // with --full-path
+    let root = te.system_root();
+    let prefix = escape(&root.to_string_lossy());
+    // before prune
+    te.assert_output(
+        &[
+            "--full-path",
+            &format!("^{prefix}.*zap.*foo$", prefix = prefix),
+        ],
+        "bar/zap/c.foo
+         bar/zap/foo/d.foo
+         bar/zap/foo",
+    );
+    // after prune
+    te.assert_output(
+        &[
+            "--prune",
+            "--full-path",
+            &format!("^{prefix}.*zap.*foo$", prefix = prefix),
+        ],
+        "bar/zap/c.foo
+         bar/zap/foo",
+    );
+
+    // --type f
+    te.assert_output(&["--prune", "--type", "f", "foo"], "bar/zap/c.foo");
+    // --type d
+    te.assert_output(
+        &["--prune", "--type", "d", "foo"],
+        "bar/zap/foo
+        foo",
+    );
+    // --prune with other options but no pattern
+    te.assert_output(
+        &["--prune", "--type", "d"],
+        "bar/zap/foo
+         bar/zap
+         bar
+         foo",
+    );
+
+    // --extention
```

Comment:
typo

Reply:
also, I think extension tests would be similar to `--type` tests. not sure if we need tests for both.

Reply:
that is true... i can remove the extension tests

Reply:
removed in commit 3